Swift_Basics/main.swift

```diff
@@ -7,5 +7,490 @@
 
 import Foundation
 
-print("Hello, World!")
+enum CarAction {
+     case stopEngine
+     case startEngine
+     case openWindow
+     case closeWindow
+     case openDoor
+     case closeDoor
+     case loadTrunk(volume: Int)
+     case unloadTrunk(volume: Int)
+     case drive(distance: Int)
+     case color(to: Color)
+     case carInfo
+     case speed(up: Int)
+     case refuel(volume: Int)
+     case pumpup
+ }
 
+ enum EngineState {
+     case started
+     case stoped
+ }
+
+ enum WindowsState {
+     case opened
+     case closed
+ }
+
+enum DoorState {
+    case opened
+    case closed
+}
+
+enum CarcassType {
+    case sport
+    case truck
+}
+
+enum TransmissionType {
+    case auto
+    case manual
+}
+
+enum Color: String {
+    case white = "white"
+    case black = "black"
+    case red = "red"
+    case blue = "blue"
+    case purple = "purple"
+ }
+
+class Engine {
+    let maxSpeed: Int
+    let fuelСonsumption:Int
+    let fuelTank: FuelTank
+    private (set) var currentSpeed: Int
+    private (set) var state: EngineState {
+        didSet {
+            print("Engine was \(self.state)")
+        }
+    }
+    
+    init(maxSpeed: Int, fuelСonsumption: Int, fuelTank: FuelTank) {
+        self.maxSpeed = maxSpeed
+        self.fuelСonsumption = fuelСonsumption
+        self.fuelTank = fuelTank
+        self.state = .stoped
+        self.currentSpeed = 0
+    }
+    
+    func change(state: EngineState) {
+        self.state = state
+    }
+    
+    func speed(up speed: Int) {
+        guard speed > 0 else { print("you cannot have negative speed less"); return }
```

Comment:
```suggestion
        guard speed > 0 else { print("you cannot have negative speed less") return }
```
стыдно такие ошибки делать